.editorconfig

```diff
@@ -23,6 +23,32 @@ insert_final_newline = true
 indent_size = 4
 dotnet_sort_system_directives_first = true
 
+# Don't use this. qualifier
+dotnet_style_qualification_for_field = false:suggestion
+dotnet_style_qualification_for_property = false:suggestion
+
+# use int x = .. over Int32
+dotnet_style_predefined_type_for_locals_parameters_members = true:suggestion
+
+# use int.MaxValue over Int32.MaxValue
+dotnet_style_predefined_type_for_member_access = true:suggestion
+
+# Require var all the time.
+csharp_style_var_for_built_in_types = true:suggestion
+csharp_style_var_when_type_is_apparent = true:suggestion
+csharp_style_var_elsewhere = true:suggestion
+
+# Disallow throw expressions.
+csharp_style_throw_expression = false:suggestion
```

Comment:
Can any (better many) of these work without `:suggestion`❔ I realize a few rules e.g. `csharp_style_var_elsewhere` may need to be easily overridden but hope that's not the case for most.

Reply:
Are you proposing something stronger (i.e. warning), or something weaker (i.e. no indication in the IDE at all)?

Reply:
`warning` or `true` (I don't remember the distinction offhand)

Reply:
`true/false` is whether to prefer these, or prefer the opposite.  The thing after the `:` is the enforcement level:

`none, silent, suggestion, warning, error`.

Reply:
So simple😃 My preference is `true:warning` and `false:warning` wherever possible i.e. unless it causes lots 'o churn in our codebase.

Reply:
Since we have `<TreatWarningsAsErrors>true</>` that ends up being an error anyway, which is pretty strong. It means your build fails and you can't even run tests if you have that wrong. Maybe we'll get there, but I have a feeling people might not like it if we go straight to errors in one go.

Reply:
That makes sense. At some point when I'm not busy, I might try changing a few of these to see if everything builds w/o making changes in the sources.

Reply:
@Pilchie Warnings as errors should be in CI only.

If it's not already in CI only, I'd suggest doing so. This will give developers more flexibility running tests while warnings are there, and will also ensure that no warnings goes into the `main` branch.

Reply:
> Warnings as errors should be in CI only.

This repo is not configured that way. For one thing, we don't want the CI to surprise users with warnings they didn't see locally.

Reply:
I've used it both ways, and typically prefer the WarnAsError on CI, but I don't think this PR is the place to change it for the repo.  Something we can talk about at @dotnet/aspdoi maybe and get more opinions on.